pkg/reconciler/v1alpha1/route/reconcile_resources.go

```diff
@@ -70,6 +67,22 @@ func (c *Reconciler) getClusterIngressForRoute(route *v1alpha1.Route) (*netv1alp
 	return ingresses[0], nil
 }
 
+func routeOwnerLabelSelector(route *v1alpha1.Route) labels.Selector {
+	return labels.Set(map[string]string{
+		serving.RouteLabelKey:          route.Name,
+		serving.RouteNamespaceLabelKey: route.Namespace,
+	}).AsSelector()
+}
+
+func (c *Reconciler) deleteClusterIngressesForRoute(route *v1alpha1.Route) error {
+	selector := routeOwnerLabelSelector(route).String()
+
+	// We always use DeleteCollection because even with a fixed name, we apply the labels.
+	return c.ServingClientSet.NetworkingV1alpha1().ClusterIngresses().DeleteCollection(
```

Comment:
If I read things correctly, this might return a NotFound if things are not there? Just jotting down here, because it helps me understand below...

Reply:
if I intentionally bust the label selector on DeleteCollection so that it selects nothing, it removes the finalizer (and leaves the ClusterIngress).  So I think we're good.